### PR TITLE
feat: add "all list" command to list all available resources

### PR DESCRIPTION
**Pull request by pauhull**
As described in #545

### CODE REVIEW

File: internal/cmd/all/list.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+var listCmd = base.Cmd{
+	BaseCobraCommand: func(client hcapi2.Client) *cobra.Command {
+		cmd := &cobra.Command{
+			Use:   "list FLAGS",
+			Short: "List all resources",
+		}
+
+		cmd.Flags().StringP("selector", "l", "", "Selector to filter by labels")
+
+		cmd.Flags().Bool("paid", false, "Only list paid resources")
```
Comment (apricote) [inline internal/cmd/all/list.go:40]: Maybe this is easier to understand?

```suggestion
		cmd.Flags().Bool("paid", false, "Only list resources that cost money")
```

Comment (jooola) [inline internal/cmd/all/list.go:40]: If I may shim in here: what does `paid` refer to ? Is that to exclude resources that were not created by the user ? e.g. official images/isos ?

File: internal/cmd/all/list.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+		for i, lc := range cmds {
+			i, lc := i, lc
+			ch := make(chan response)
+			responseChs[i] = ch
+
+			go func() {
+				defer close(ch)
+
+				labelSelector, _ := cmd.Flags().GetString("selector")
```
Comment (apricote) [inline internal/cmd/all/list.go:94]: Could be moved to line 49 to avoid reparsing the argument for every resource

File: internal/cmd/all/list.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+				flagSet := pflag.NewFlagSet(lc.JSONKeyGetByName, pflag.ExitOnError)
+
+				switch lc.JSONKeyGetByName {
+				case image.ListCmd.JSONKeyGetByName:
+					flagSet.StringSlice("type", []string{"backup", "snapshot"}, "")
+				case iso.ListCmd.JSONKeyGetByName:
+					flagSet.StringSlice("type", []string{"private"}, "")
+				}
+
+				_ = flagSet.Parse([]string{})
```
Comment (apricote) [inline internal/cmd/all/list.go:108]: Can you add a comment why this is necessary? To the unknowing reader this looks like a no-op.

File: internal/cmd/all/list.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+				result, err := lc.Fetch(ctx, client, flagSet, listOpts, []string{})
+				ch <- response{result, err}
+			}()
+		}
+
+		resources := make([][]any, len(cmds))
```
Comment (apricote) [inline internal/cmd/all/list.go:114]:
```suggestion
    // Wait for all requests to finish and collect results
		resources := make([][]any, len(cmds))
```

**Review by apricote — CHANGES_REQUESTED** [review]

File: internal/cmd/all/list.go (hunk 1 above)
Comment (jooola) [reply internal/cmd/all/list.go:40]: I think introducing the notion of paid with this flag might lead to confusions.

I would rather use another flag name (e.g --owned) or invert it with --include-read-only-resource (yeah this is too long, but maybe you folks have an idea?). Using a --all flag might be a bit repetitive `all list --all`.

Comment (apricote) [reply internal/cmd/all/list.go]: What would be the usecase to list all static resources? I think the user would rather like to see their resources.

---

I think we are conflating two discussions:

- _Paid_ (servers, IPs, LBs, Volumes) vs. _free_ (Network, SSH Key, Placement Group)
- _Project Resouces_ (Server, LB, PG, Snapshots+Backups Images) vs _Static/Public_ (System+App Images, Public ISOs, Location, Network Zone, Server Types, Load Balancer Types)

Comment (apricote) [reply internal/cmd/all/list.go]: We have a similar pattern in `console.hetzner.cloud` on the Project Card View. Here it shows all paid resources as icons for every project and unpaid resources are grouped under "Other":

screenshot-paid-unpaid

Comment (jooola) [reply internal/cmd/all/list.go]: Oh alright, so list all is only for project resources ? In that case, the paid flag name make more sense. But this isn't This wasn't clear from the current description of the command, I would have expected to list all resource (+ static/public)

Comment (pauhull) [reply internal/cmd/all/list.go]: The long description now explains the command in more detail and explicitly states that static and public resources are not included.

**Review by apricote — APPROVED** [review]

**Pull request merged**